core/src/main/scala/kafka/server/Kip500Broker.scala

```diff
@@ -111,7 +111,10 @@ class Kip500Broker(val config: KafkaConfig,
   var quotaManagers: QuotaFactory.QuotaManagers = null
 
   val brokerMetaPropsFile = "meta.properties"
-  val brokerMetadataCheckpoints = config.logDirs.map(logDir => (logDir, new BrokerMetadataCheckpoint(new File(logDir + File.separator + brokerMetaPropsFile)))).toMap
+
+  // Look for metadata checkpoints (meta.properties) in all log.dirs and metadata.log.dir
+  val brokerMetadataLogDirs = config.logDirs:+(config.metadataLogDir)
```

Comment:
I'd really prefer just to use an "if" statement here.  The :+ operator is pretty obscure and hard to call up documentation for

Reply:
Yea, fair enough.